### PR TITLE
GH-213 Merge Binder SPI into Core

### PR DESCRIPTION
**Pull request by mbogoevici**
- Moved contents of `spring-cloud-stream-binder-spi` into `spring-cloud-stream`
- Removed `spring-cloud-stream-binder-local` and used local binder implementation for tests exclusively

This resolves spring-cloud/spring-cloud-stream#213

### CODE REVIEW

File: spring-cloud-stream-binders/spring-cloud-stream-binder-kafka/src/test/java/org/springframework/cloud/stream/binder/kafka/RawModeKafkaBinderTests.java [hunk 1]
```diff
@@ -54,6 +54,7 @@
  * @author Gary Russell
  */
 
+@Ignore
```
Comment (markfisher) [inline spring-cloud-stream-binders/spring-cloud-stream-binder-kafka/src/test/java/org/springframework/cloud/stream/binder/kafka/RawModeKafkaBinderTests.java]: is this addition intentional?

Comment (mbogoevici) [inline spring-cloud-stream-binders/spring-cloud-stream-binder-kafka/src/test/java/org/springframework/cloud/stream/binder/kafka/RawModeKafkaBinderTests.java]: Yes, the Kafka tests have been failing for a while after the Rabbit PubSub changes - need to address that separately.

Comment (mbogoevici) [inline spring-cloud-stream-binders/spring-cloud-stream-binder-kafka/src/test/java/org/springframework/cloud/stream/binder/kafka/RawModeKafkaBinderTests.java]: The failures have been sporadic, though (and unrelated to the PR).

File: spring-cloud-stream/pom.xml [hunk 2]
```diff
@@ -36,29 +36,30 @@
...
 		<dependency>
-			<groupId>org.springframework.boot</groupId>
-			<artifactId>spring-boot-starter-test</artifactId>
-			<scope>test</scope>
+			<groupId>com.fasterxml.jackson.core</groupId>
+			<artifactId>jackson-databind</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>org.apache.httpcomponents</groupId>
+			<artifactId>httpclient</artifactId>
```
Comment (markfisher) [inline spring-cloud-stream/pom.xml]: why is this dependency needed in the core now?

Comment (mbogoevici) [inline spring-cloud-stream/pom.xml]: It was in the SPI. We can remove it, I believe - it's an offshoot of the RabbitAdmin being present in the Binder SPI at some point.

File: spring-cloud-stream-binders/spring-cloud-stream-binder-kafka/src/test/java/org/springframework/cloud/stream/binder/kafka/RawModeKafkaBinderTests.java (hunk 1 above)
Comment (markfisher) [inline spring-cloud-stream-binders/spring-cloud-stream-binder-kafka/src/test/java/org/springframework/cloud/stream/binder/kafka/RawModeKafkaBinderTests.java]: okay, thanks

File: spring-cloud-stream/pom.xml (hunk 2 above)
Comment (markfisher) [inline spring-cloud-stream/pom.xml]: I will try to remove it. If the build passes I will merge without it.

Comment (mbogoevici) [inline spring-cloud-stream/pom.xml]: Ok, cool - thanks.

Comment (markfisher) [inline spring-cloud-stream/pom.xml]: The build passes with it removed. Actually, the same goes for the jackson dependency. So, I am removing both on merge.
